js/about/passwords.js

```diff
@@ -63,7 +63,7 @@ class SiteItem extends React.Component {
   }
 
   onDelete () {
-    aboutActions.deletePasswordSite(this.props.site)
+    aboutActions.deletePassword(this.props.site.toJS())
```

Comment:
This is supposed to undo the 'never save password' site setting, not delete the password itself